lib/std/process.zig

```diff
@@ -53,9 +52,200 @@ test "getCwdAlloc" {
     testing.allocator.free(cwd);
 }
 
-/// Caller owns resulting `BufMap`.
-pub fn getEnvMap(allocator: Allocator) !BufMap {
-    var result = BufMap.init(allocator);
+pub const EnvMap = struct {
+    hash_map: HashMap,
+
+    const HashMap = std.HashMap(
+        []const u8,
+        []const u8,
+        EnvNameHashContext,
+        std.hash_map.default_max_load_percentage,
+    );
+
+    pub const Size = HashMap.Size;
+
+    pub const EnvNameHashContext = struct {
+        fn upcase(c: u21) u21 {
+            if (c <= std.math.maxInt(u16))
+                return std.os.windows.ntdll.RtlUpcaseUnicodeChar(@intCast(u16, c));
+            return c;
+        }
+
+        pub fn hash(self: @This(), s: []const u8) u64 {
+            _ = self;
+            if (builtin.os.tag == .windows) {
+                var h = std.hash.Wyhash.init(0);
+                var it = std.unicode.Utf8View.initUnchecked(s).iterator();
```

Comment:
Using `initUnchecked` in the hash context without validating the key in `put`/`get` is a really bad idea as `Utf8Iterator` does a whole bunch of `catch unreachable`.

Reply:
This is intentional because the utf8 string in this map come from decoding utf16 strings on Windows.

Reply:
`child_process.env_map.put(user_provided_key, some_value)` seems like something you could find. It should at least be documented that on Windows the keys must be valid utf-8.

Reply:
Ok I've documented that the `key` argument for all functions must be valid UTF8 on windows.